### PR TITLE
chore(SwingSet): Consolidate logic for picking a queued event

### PR DESCRIPTION
**Pull request by gibson042**
## Description

Cleanup of a part of the kernel implementation that I found confusing.

### Security Considerations

n/a

### Scaling Considerations

n/a

### Documentation Considerations

I don't think this affects any documentation.

### Testing Considerations

n/a

### CODE REVIEW

**Review by mhofman — APPROVED** [review]
Looks fine by me, it sure is less verbose and less convoluted. I'd let @warner chime in too.

File: packages/SwingSet/src/kernel/gc-actions.js [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
 
-const typePriority = ['dropExport', 'retireExport', 'retireImport'];
+const { ownKeys } = Reflect;
+
+/**
+ * A descending-priority mapping of GC action type to queue event type.
+ *
+ * @enum {string}
+ */
+const typePriority = {
```
Comment (warner) [inline packages/SwingSet/src/kernel/gc-actions.js]: I'm not against this change from list to object/record, but.. is it a win? Seems like it's adding more code than it's removing, and the need for `setPrototypeOf(.. null)` is a smell. If it helps us detect typos in the `dropExport`/etc strings, then I can see it being a benefit, but I think we had that already (at the cost of an O(3) lookup).

File: packages/SwingSet/src/kernel/kernel.js [hunk 2]
```diff
@@ -1645,39 +1643,28 @@ export default function buildKernel(
...
+        message: acceptanceMessage,
+        processor: processAcceptanceMessage,
+      };
     }
-
-    return { message, processor };
+    const message =
+      processGCActionSet(kernelKeeper) ||
+      kernelKeeper.nextReapAction() ||
+      kernelKeeper.getNextRunQueueMsg();
```
Comment (warner) [inline packages/SwingSet/src/kernel/kernel.js:1728]: you might go ahead and delete `kernelKeeper.isRunQueueEmpty` too

**Review by warner — APPROVED** [review]
works for me

File: packages/SwingSet/src/kernel/gc-actions.js (hunk 1 above)
Comment (FUDCo) [reply packages/SwingSet/src/kernel/gc-actions.js]: I concur.  This seems more convoluted without compensating benefit.

Comment (gibson042) [reply packages/SwingSet/src/kernel/gc-actions.js]: The smell I'm addressing here is ``` `${type}s` ```, which isn't a typo hazard as much as it is a grep dodge... better to explicitly show the map from one domain to another in my opinion. And the `setPrototypeOf` isn't necessary, just a way to be explicit (a reasonable alternative would omit it and replace later use of `in` with an own property check).

That said, I can revert back to the array if you feel better about that. Thoughts?

File: packages/SwingSet/src/kernel/kernel.js (hunk 2 above)
Comment (gibson042) [reply packages/SwingSet/src/kernel/kernel.js:1728]: Done, along with `isAcceptanceQueueEmpty`.

File: packages/SwingSet/src/kernel/gc-actions.js (hunk 1 above)
Comment (warner) [reply packages/SwingSet/src/kernel/gc-actions.js]: Agreed that `${type}s` is a smell.

Using a record or Map as the primary list doesn't make it clear that the priority ordering of these three types is critical. Your implementation does the right thing, but (speaking from personal experience) most programmers/auditors don't automatically know that property enumeration happens in insertion order.

So how about:

```js
// priority of GC action types
const priorities = ['dropExport', 'retireExport', 'retireImport'];

// map GC action type to queue event type
const actionTypeToQueueEventType = new Map([
  [ 'dropExport', 'dropExports' ],
  [ 'retireExport', 'retireExports' ],
  [ 'retireImport', 'retireImports' ],
]);

...

assert(actionTypeToQueueEventType.has(type), type);

..

for (const type of priorities) {

..

const queueType = actionTypeToQueueEventType.get(type);
return harden({ type: queueType, ..
```

**Review by warner — APPROVED** [review]
conditional approval with the recommended `actionTypeToQueueEventType` change

**Pull request merged**